### DISCUSSION TITLE
Access per-page config in custom App?

### DISCUSSION THREAD

**Discussion by jpvanoosten** [vercel/next.js#16214] [Help] [answered]
A page can export a `config` object that contains some per-page configuration options. The only example of this that I can find is specifying the "amp" config for a page:

```javascript
export const config = { amp: true }
```
 
I have two questions about this:

1. Is it possible to add custom configuration to the `config` object exported on the page?
2. Is it possible to read the exported `config` object in a custom App (in `/page/_app.{js|ts}`?

Use case: In order to configure the app layout, I would like to specify per-page config options for a top-level `Layout` component that wraps all page components (using a custom App).

**Comment by jamesmosier** [comment, marked as answer]
There is no such custom per-page configuration built into Next currently.

Could you do something like this:

```jsx
// pages/my-page.js

export default function MyPage() {
  return <div/>
}

MyPage.customConfig = {
  property1: 123
}
```

```jsx
// _app

function MyApp({ Component, pageProps }) {
  const myProperty = Component.customConfig.property1

  return <Component {...pageProps} />
}

```

**Reply by jpvanoosten** [reply]
@jamesmosier Thanks! I found a similar solution elsewhere but I was hoping to avoid the static properties on the exported page function, but I just wanted to check!

Thanks for the response!

**Comment by abhilashlr7** [comment]
Curious, whats the equivalent of the same in App router? Like if we want to set some configs at the root Page level, and use it as part of say a nested page.tsx, like metadata nesting, similarly if we would want to do something similar for breadcrumbs for example. Is there a way to achieve it?
